vllm/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -269,8 +252,31 @@ def all_reduce_unreg(self, inp: torch.Tensor, out: torch.Tensor = None):
         custom_ar.all_reduce_unreg(self._ptr, inp, self.buffer, out)
         return out
 
+    def custom_all_reduce(self, input: torch.Tensor) -> Optional[torch.Tensor]:
+        # when custom allreduce is disabled, this will be None
+        if self.disabled:
+            return None
+        if self._IS_CAPTURING:
+            if torch.cuda.is_current_stream_capturing():
+                if self.should_custom_ar(input):
+                    return self.all_reduce_reg(input)
+            else:
+                if self.should_custom_ar(input):
+                    # if warm up, mimic the allocation pattern
+                    # since custom allreduce is out-of-place
+                    return torch.empty_like(input)
+        else:
+            # note: outside of cuda graph context,
+            # custom allreduce incurs a cost of cudaMemcpy, which should
+            # be small(<=1% of overall latency) compared to the performance
+            # gains of using custom kernels
+            if self.should_custom_ar(input):
+                return self.all_reduce_unreg(input)
+
+        return None
+
     def close(self):
-        if self._ptr:
+        if not self.disabled and self._ptr:
```

Comment:
```suggestion
        if custom_ar is not None and self._ptr:
```

Reply:
fixed in [dd71361](https://github.com/vllm-project/vllm/pull/4754/commits/dd71361dead3d295aa2c63a2c029a3d57c047fbc).  `custom_ar is None` implies `self.disabled == True`